packages/compiler/src/server/lib-provider.ts

```diff
@@ -57,8 +57,11 @@ export class LibraryProvider {
     const data = await pkg.getPackageJsonData();
     // don't add to cache when failing to load package.json which is unexpected
     if (!data) return false;
+    // Modification Note:
+    // - devDependencies are theoretically not counted in the production environment;
+    // - this leads to LSP connection failure after dynamically loading a certain library in the package.
```

Comment:
1. remove the 'modification note' which you can put in comit comment but will be confusing in the code without the context
2. please update the comment:
// Only check dependencies and peerDependencies for @typespec/compiler and exclude devDependencies to further limit the libraries to load and avoid potential issues cause by loading random libraries. 
// TODO: add more filter condition when needed